lib/external-grading/file-store-disk.js

```diff
@@ -0,0 +1,54 @@
+const ERR = require('async-stacktrace');
+const async = require('async');
+const path = require('path');
+const fs = require('fs-extra');
+const tar = require('tar');
+const tmp = require('tmp');
+
+const externalGraderCommon = require('./common');
+
+module.exports.createFilesForJob = function(grading_job, submission, variant, question, course, callback) {
+
+    const storageDir = externalGraderCommon.getJobDirectory(grading_job.id);
+    let dir;
+    let cleanup;
+
+    async.series([
+        (callback) => {
+            tmp.dir({ unsafeCleanup: true }, (err, path, cleanupCallback) => {
+                if (ERR(err, callback)) return;
+                dir = path;
+                cleanup = cleanupCallback;
+                callback(null);
+            });
+        },
+        (callback) => {
+            externalGraderCommon.buildDirectory(dir, submission, variant, question, course, callback);
```

Comment:
Although it's clearly longer and more horrible, I prefer to do the explicit callback here with `ERR(...)` so that we get a proper stacktrace on errors. Hopefully we can make this all go away with async/await soon!